etc/categoryedit

```diff
@@ -9,6 +9,12 @@ function error {
   exit 1
 }
 
+#Only one instance of this script should run at a time. wait if another process other than this one is running
+while [[ $(pgrep categoryedit | sort |  head -n 1) != $$ ]]; do
+  # waiting for another instance of categoryedit to finish
+  sleep 0.02
+done
+
```

Comment:
hmm.. I'll have to think about what should be done in this case. (will get back to it in a few hours)

Reply:
could we break out the GUI part of this file into its own file? that way the Category editor could just source `etc/categoryedit` to obtain the necessary functions and we can still use it here like this

Reply:
> could we break out the GUI part of this file (basically the stuff in the while loop) into its own file? that way the Category editor could just source `etc/categoryedit` to obtain the necessary functions and we can still use it here like this

Your suggestion gave me a different idea. There could be a category function in `api` that categoryedit then uses. The category function would contain a waiting function.

Ont other way to solve this would be to implement a lockfile.

Reply:
> Your suggestion gave me a different idea. There could be a category function in `api` that categoryedit then uses. The category function would contain a waiting function.

well the waiting function in use now wouldn't work, since it relies on the PID of the categoryedit file (which is different from the script that spawned it). I'd rather not use a lockfile since that could result in somehow being permanently locked while I don't think that could really ever happen with the PID check

Reply:
@Botspot so... I solved this but it didn't need to be solved. you can't have the Category Editor running in the background. the GUI won't load till you exit it, it is blocking 😆 

I have pushed my changes anyway just so you can see what it took to filter out the GUI vs non-gui launch of categoryeditor https://github.com/Botspot/pi-apps/pull/1932/commits/74f0293167f101b031b939be97ffee4f4d848e05

Reply:
also keep in mind, you can't test out this standalone without an instance of categoryedit running
```
pgrep categoryedit | xargs ps -o pid,cmd h | awk '{print $1, $4}' | awk '$2 != ""' | awk '{print $1}' | sort | head -n 1
```
otherwise it produces false output (it outputs all PIDs since nothing gets piped to xargs)

this is fine because there is always be at least one PID found, since this is running in categoryedit itself